report/node.go

```diff
@@ -182,8 +184,26 @@ func (n Node) WithChildren(children NodeSet) Node {
 
 // WithChild returns a fresh copy of n, with one child merged in.
 func (n Node) WithChild(child Node) Node {
-	n.Children = n.Children.Merge(MakeNodeSet(child))
-	return n
+	result := n.Copy()
+	result.Children = result.Children.Merge(MakeNodeSet(child))
+	return result
+}
+
+// Copy returns a value copy of the Node.
+func (n Node) Copy() Node {
```

Comment:
> part of an old commit

Are you referring to 9e092f1a4? That looks quite different, and doesn't have these `Copy()` changes. Looks to me like a re-base has gone wrong on this branch, perhaps.